catalog/cataloger_merge.go

```diff
@@ -11,10 +11,14 @@ import (
 )
 
 const (
-	MergeBatchSize = 256
+	MergeBatchSize       = 256
+	MergeBatchChanBuffer = 10
 )
 
-type mergeBatchRecords []*diffResultRecord
+type mergeBatchRecords struct {
```

Comment:
why use channel of batch operations - we have diff iterators that produce a channel of changes, we can batch them as we did before (if we like) - we already as part of one big transaction, so the batching can be done per kind of operation on the merge level.